ui/src/View.elm

```diff
@@ -286,15 +291,234 @@ buildUrl { build, source, target } =
             |> urlFromRoute
 
 
-recordView : Filters -> BuildRecord -> Html Msg
-recordView filters ({ id, build, download, source, target, systemAddons } as record) =
+buildBuildIdUrl : String -> String
+buildBuildIdUrl buildId =
+    { product = []
+    , version = []
+    , platform = []
+    , channel = []
+    , locale = []
+    , buildId = buildId
+    , search = ""
+    , page = 1
+    }
+        |> routeFromFilters
+        |> urlFromRoute
+
+
+onClick_ : msg -> Attribute msg
+onClick_ msg =
+    onWithOptions
+        "click"
+        { preventDefault = True, stopPropagation = True }
+        (Decode.succeed msg)
+
+
+recordView : Filters -> List String -> BuildRecord -> Html Msg
+recordView filters expanded ({ id, build, download, source, target, systemAddons } as record) =
     let
         filename =
             String.split "/" download.url
                 |> List.reverse
                 |> List.head
                 |> Maybe.withDefault ""
 
+        ( revision, revisionUrl ) =
+            case ( source.repository, source.revision ) of
+                ( Just repository, Just revision ) ->
+                    ( revision, Just <| repository ++ "/rev/" ++ revision )
+
+                _ ->
+                    ( "unknown", Nothing )
+
+        sourceTree =
+            Maybe.withDefault "unknown" source.tree
+
+        buildDate =
+            case build of
+                Just { date } ->
+                    date
+
+                Nothing ->
+                    download.date
+
+        isExpanded =
+            List.member id expanded
+    in
+        tbody []
+            [ tr
+                [ Html.Attributes.id id ]
+                [ td []
+                    [ a
+                        [ href <| buildPermalink record
+                        , title "Build permalink"
+                        ]
+                        [ i [ class "glyphicon glyphicon-link" ] [] ]
+                    ]
+                , td []
+                    [ a
+                        [ href "#"
+                        , onClick_ (ToggleBuildDetails id)
+                        , title "View build details"
+                        ]
+                        [ i
+                            [ class <|
+                                "glyphicon glyphicon-eye-"
+                                    ++ (if isExpanded then
+                                            "close"
+                                        else
+                                            "open"
+                                       )
+                            ]
+                            []
+                        ]
+                    ]
+                , td [ title source.product ]
+                    [ highlighSearchTerm filters.search filters.product source.product ]
```

Comment:
Just realised that there's a typo in the name, it's missing a "t": `highlightSearchTerm`